actors/builtin/storage_power/storage_power_actor.go

```diff
@@ -159,69 +203,90 @@ func (a *StoragePowerActor) DeleteMiner(rt Runtime, minerAddr addr.Address) *adt
+	if params.terminationType != builtin.NormalExpiration {
+		amountToSlash := pledgePenaltyForSectorTermination(params.weight, params.terminationType)
 		a.slashPledgeCollateral(rt, minerAddr, amountToSlash)
 	}
 	return &adt.EmptyValue{}
 }
 
-func (a *StoragePowerActor) OnSectorTemporaryFaultEffectiveBegin(rt Runtime, weight autil.SectorStorageWeightDesc) *adt.EmptyValue {
+type OnSectorTemporaryFaultEffectiveBegin struct {
+	weight autil.SectorStorageWeightDesc
+}
+
+func (a *StoragePowerActor) OnSectorTemporaryFaultEffectiveBegin(rt Runtime, params *OnSectorTemporaryFaultEffectiveBegin) *adt.EmptyValue {
 	rt.ValidateImmediateCallerType(builtin.StorageMinerActorCodeID)
-	if err := a.deductClaimedPowerForSector(rt, rt.ImmediateCaller(), weight); err != nil {
+	if err := a.deductClaimedPowerForSector(rt, rt.ImmediateCaller(), params.weight); err != nil {
 		rt.Abort(exitcode.ErrIllegalState, "Failed to deduct claimed power for sector: %v", err)
 	}
 	return &adt.EmptyValue{}
 }
 
-func (a *StoragePowerActor) OnSectorTemporaryFaultEffectiveEnd(rt Runtime, weight autil.SectorStorageWeightDesc) *adt.EmptyValue {
+type OnSectorTemporaryFaultEffectiveEnd struct {
+	weight autil.SectorStorageWeightDesc
+}
+
+func (a *StoragePowerActor) OnSectorTemporaryFaultEffectiveEnd(rt Runtime, params *OnSectorTemporaryFaultEffectiveEnd) *adt.EmptyValue {
 	rt.ValidateImmediateCallerType(builtin.StorageMinerActorCodeID)
-	if err := a.addPowerForSector(rt, rt.ImmediateCaller(), weight); err != nil {
+	if err := a.addPowerForSector(rt, rt.ImmediateCaller(), params.weight); err != nil {
 		rt.Abort(exitcode.ErrIllegalState, "Failed to add power for sector: %v", err)
 	}
 	return &adt.EmptyValue{}
 }
 
-func (a *StoragePowerActor) OnSectorModifyWeightDesc(
-	rt Runtime, prevWeight autil.SectorStorageWeightDesc, newWeight autil.SectorStorageWeightDesc) *adt.EmptyValue {
+type OnSectorModifyWeightDesc struct {
+	prevWeight autil.SectorStorageWeightDesc
+	newWeight  autil.SectorStorageWeightDesc
+}
+
+func (a *StoragePowerActor) OnSectorModifyWeightDesc(rt Runtime, params *OnSectorModifyWeightDesc) *adt.EmptyValue {
 
 	rt.ValidateImmediateCallerType(builtin.StorageMinerActorCodeID)
-	if err := a.deductClaimedPowerForSector(rt, rt.ImmediateCaller(), prevWeight); err != nil {
+	if err := a.deductClaimedPowerForSector(rt, rt.ImmediateCaller(), params.prevWeight); err != nil {
 		rt.Abort(exitcode.ErrIllegalState, "Failed to deduct claimed power for sector: %v", err)
 	}
-	if err := a.addPowerForSector(rt, rt.ImmediateCaller(), newWeight); err != nil {
+	if err := a.addPowerForSector(rt, rt.ImmediateCaller(), params.newWeight); err != nil {
 		rt.Abort(exitcode.ErrIllegalState, "Failed to add power for sector: %v", err)
 	}
 	return &adt.EmptyValue{}
 }
 
-func (a *StoragePowerActor) OnMinerSurprisePoStSuccess(rt Runtime) *adt.EmptyValue {
+func (a *StoragePowerActor) OnMinerSurprisePoStSuccess(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
```

Comment:
To keep method signatures (`func (Runtime, CBORMarshal)`) the same I've introduced this parameter, any objection?